docs/source/install.rst

```diff
@@ -240,10 +285,96 @@ Tests are run with::
 The test suite requires that the proj-datumgrid package is installed
 in :envvar:`PROJ_LIB`.
 
-If tests are not required, PROJ can be built without the test suite using the following configuration::
 
-    cmake -DBUILD_TESTING=OFF ..
+CMake configure options
++++++++++++++++++++++++
+
+Options to configure a CMake are provided using ``-D<var>=<value>``.
+All cached entries can be viewed using ``cmake -LAH`` from a build directory.
+
+.. option:: BUILD_SHARED_LIBS
+
+    Build PROJ library shared. Default for Windows is OFF, building only
+    a static library. Default for all others is ON.
+
+    .. versionchanged:: 7.0
+        Renamed from ``BUILD_LIBPROJ_SHARED``
+
+.. option:: BUILD_TESTING=ON
+
+    Build the testing tree. Default is ON, but can be turned OFF if tests
+    are not required.
+
+    .. versionchanged:: 7.0
+        Renamed from ``PROJ_TESTS``
+
+.. option:: CMAKE_BUILD_TYPE
+
+   Choose the type of build, options are: None (default), Debug, Release,
+   RelWithDebInfo, or MinSizeRel.
+
+   .. note::
+       A default build is not optimized without specifying
+       ``-DCMAKE_BUILD_TYPE=Release`` (or similar) during configuration,
+       or by specifying ``--config Release`` with CMake
+       multi-configuration build tools (see example below).
+
+.. option:: CMAKE_C_COMPILER
+
+    C compiler. Ignored for some generators, such as Visual Studio.
+
+.. option:: CMAKE_C_FLAGS
+
+    Flags used by the C compiler during all build types.
+
+.. option:: CMAKE_CXX_COMPILER
+
+    C++ compiler. Ignored for some generators, such as Visual Studio.
+
+.. option:: CMAKE_CXX_FLAGS
+
+    Flags used by the C++ compiler during all build types.
+
+.. option:: CMAKE_INSTALL_PREFIX
+
+    Default for Windows is based on the environment variably
+    :envvar:`OSGEO4W_ROOT` (if set), otherwise is ``c:/OSGeo4W``.
+    Default for Unix-like is ``/usr/local/``.
+
+.. option:: ENABLE_CURL=ON
+
+    Enable CURL support, default ON.
+
+.. option:: CURL_INCLUDE_DIR
+
+    Path to an include directory with the ``curl`` directory.
+
+.. option:: CURL_LIBRARY
+
+    Path to a shared or static library file, such as ``curl.dll``,
+    ``libcurl.so``, or other name.
+
+.. option:: ENABLE_IPO=OFF
+
+    Build library using the compiler's `interprocedural optimization
+    <https://en.wikipedia.org/wiki/Interprocedural_optimization>`_
+    (IPO), if available, default OFF.
+
+    .. versionchanged:: 7.0
+        Renamed from ``ENABLE_LTO``.
+
+.. option:: EXE_SQLITE3
+
+    Path to an ``sqlite3`` or ``sqlite3.exe`` executable.
+
+.. option:: SQLITE3_INCLUDE_DIR
+
+    Path to an include directory with the ``sqlite3.h`` header file.
+
+.. option:: SQLITE3_LIBRARY
 
+    Path to a shared or static library file, such as ``sqlite3.dll``,
+    ``libsqlite3.so``, ``sqlite3.lib`` or other name.
```

Comment:
for consistency TIFF_INCLUDE_DIR and TIFF_LIBRARY should also likely be documented.

Reply:
oddly there were `TIFF_LIBRARY_RELEASE` and `TIFF_LIBRARY_DEBUG` variables